bench/index.js

```diff
@@ -0,0 +1,148 @@
+const instances = require("./instances");
+const util = require("./util");
+const mini = require("minimist");
+const { adt } = require("@masaeedu/adt");
+const { Obj, Maybe, Either, Arr } = require("../src");
+const { Just, Nothing } = Maybe;
+const { Left, Right } = Either;
+
+const Command = adt({
+  RunAll: ["Maybe Path"],
+  RunSuite: ["String", "Maybe [Int]", "Maybe Path"],
+  RunSingle: ["String", "Maybe [Int]", "Maybe Path"],
+  List: ["()"],
+  Help: ["()"]
+});
+
+const { RunAll, RunSuite, RunSingle, List, Help } = Command;
+
+const parseCommand = args => {
+  const parsedArgs = (() => {
+    const named = Obj.map(k => {
+      const r = Obj.get(k)(args);
+      return r ? Just(String(r)) : Nothing;
+    })(Obj.mirror(["save-to", "name", "sizes"]));
+
+    const positional = {
+      name: args._[1] ? Just(args._[1]) : Nothing,
+      sizes: args._[2] ? Just(String(args._[2])) : Nothing
+    };
+
+    return Obj.appendWith(Maybe.alt)(named)(positional);
+  })();
+
+  const parseSaveTo = Right(parsedArgs["save-to"]);
+
+  const parseName = Maybe.match({
+    Just: Right,
+    Nothing: Left("Missing name argument")
+  })(parsedArgs.name);
+
+  const parseSingleName = Either.chain(x => {
+    const [a, b] = x.split(".");
+    return a && b ? Right([a, b]) : Left(`invalid benchmark path ${x}`);
+  })(parseName);
+
+  const parseSizes = Right(
```

Comment:
Is this a `Maybe (Maybe something)`?

Reply:
It's an `Either String (Maybe [Int])`, I changed it a bit in d339848 so that it actually makes sure it has a list of `Int`s.